src/iuse.cpp

```diff
@@ -2643,16 +2669,17 @@ int iuse::dig( player *p, item *it, bool t, const tripoint & )
         break;
     }
 
-    const int moves = dig_pit_moves( p, it, deepen );
+    digging_moves_and_byproducts moves_and_byproducts = dig_pit_moves_and_byproducts( p, it, can_deepen,
+            false );
 
-    activity_id digging_activity( "ACT_DIG" );
-    if( deepen ) {
-        digging_activity = activity_id( "ACT_DIG_DEEPEN" );
-    }
-
-    player_activity act( digging_activity, moves, -1, p->get_item_position( it ) );
+    player_activity act( activity_id( "ACT_DIG" ), moves_and_byproducts.moves, -1,
+                         p->get_item_position( it ) );
     p->assign_activity( act );
-    p->activity.placement = pnt;
+    p->activity.placement = dig_point;
+    p->activity.values.emplace_back( moves_and_byproducts.spawn_count );
+    p->activity.str_values.emplace_back( moves_and_byproducts.byproducts_item_group );
+    p->activity.str_values.emplace_back( moves_and_byproducts.result_terrain.id().str() );
+    p->activity.coords.emplace_back( deposit_point );
```

Comment:
This usually works fine, but every once in a while, it sticks the products somewhere nearby (like 12+ tiles away) instead of at the selected location.

I'm wondering if coords[0] sometimes has garbage in it instead of it being empty?

Reply:
sounds very similar to what was happening with z-level hauling before I added [this](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/game.cpp#L11448) calculation.  Could be that it is caused by being on the edge of a submap (though y'all probably know better than I).

Reply:
I don't think it's a coordinate-based edge case, I couldn't get it to trigger twice in the same location or repeatedly without restarting the game.
My guess is uninitialized something, but I couldn't find anything.

Reply:
I think I figured it out--I believe it's caused by resuming an interrupted activity. If you start digging, pick a spot to dump the byproducts, interrupt it (I used .), then move and start a new dig, it ends up placing the byproducts in a different location.

Looking at the `player_activity::can_resume_with` I can see some handling in here for other similar activities like clearing rubble to ensure they're not incorrectly resumed, so I'll update that for these digging activities.

Reply:
Yup, that was it. I updated it so that [digging can only be resumed if it's actually got the same parameters](https://github.com/CleverRaven/Cataclysm-DDA/pull/29259/files#diff-94d984ba347779127e2dfa6608462307R192).

Reply:
Pretty sure that's default behavior for resuming.  You probably should just take out the special handling of digging in that function.

Reply:
Yeah, it's not the existing default behavior, which is why I had to add the handling, otherwise it would've just worked correctly in the first place. 

The default behavior for checking whether it can be resumed only looks at a handful of properties, none of which aside from the id/name of the activity are even relevant to digging (the `position`, `index` and `targets` are all either integers or item specific).

All the activities that stuff things into `placement`, `values`, `str_values` and `coords` have to do their own check here to prevent being resumed, and their meaning is activity specific (hence why I think this code has comments advocating for `activity_handler_actors`).

I think if someone wants to tackle updating the default behavior, that's not an unreasonable thing to do, but I don't want to drag it under this umbrella.